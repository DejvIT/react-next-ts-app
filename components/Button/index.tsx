import classNames from 'classnames';
import React, { memo } from 'react';

import { ChevronRightIcon } from '@app/components';

import styles from './index.module.scss';

interface Props {
  id?: string;
  isLoading?: boolean;
  ariaLabel?: string;
  className?: string;
  style?: React.CSSProperties;
  disabled?: boolean;
  variant?: 'primary' | 'secondary' | 'tertiary' | 'icon' | 'default' | 'dark' | 'bordered' | 'link-secondary';
  size?: 'normal' | 'small' | 'large';
  tabIndex?: number;
  type?: 'button' | 'submit';
  block?: boolean;
  withChevronRight?: boolean;
  hideChildrenWhileLoading?: boolean;
  children?: React.ReactNode;
  onClick?: () => void;
  onMouseEnter?: () => void;
  onTouchStart?: () => void;
  colorHexChevron?: string;
}

export const Button: React.FunctionComponent<Props> = (props) => {
  const {
    id,
    ariaLabel,
    disabled,
    variant = 'default',
    size = 'normal',
    tabIndex,
    style,
    className,
    isLoading,
    hideChildrenWhileLoading,
    type = 'button',
    block,
    withChevronRight,
    children,
    onClick,
    onMouseEnter,
    onTouchStart,
    colorHexChevron,
  } = props;

  const getVariantClassName = () => {
    switch (variant) {
      case 'primary':
        return styles.primary;
      case 'secondary':
        return styles.secondary;
      case 'tertiary':
        return styles.tertiary;
      case 'icon':
        return styles.icon;
      case 'dark':
        return styles.dark;
      case 'bordered':
        return styles.bordered;
      case 'link-secondary':
        return styles.linkSecondary;

      default:
        return styles.link;
    }
  };

  const getSizeClassName = () => {
    switch (size) {
      case 'small':
        return styles.sizeSmall;
      case 'large':
        return styles.sizeLarge;
      default:
        return styles.sizeNormal;
    }
  };

  return (
    <button
      aria-label={ariaLabel}
      className={classNames(
        className,
        styles.btn,
        block ? styles.block : null,
        withChevronRight ? styles.withChevron : null,
        getVariantClassName(),
        getSizeClassName(),
        disabled ? styles.disabled : null,
      )}
      disabled={disabled || isLoading}
      id={id}
      onClick={onClick}
      onMouseEnter={onMouseEnter}
      onTouchStart={onTouchStart}
      role="button"
      style={style}
      tabIndex={tabIndex}
      type={type}
    >
      {isLoading && (
        <LoadingIcon
          colorHex={
            variant === 'secondary'
              ? 'var(--clrDarkSecondary)'
              : variant === 'primary'
              ? 'var(--clrDisabledText)'
              : undefined
          }
        />
      )}

      {!hideChildrenWhileLoading && !isLoading && (
        <>
          {children}

          {withChevronRight && <ChevronRightIcon className={styles.chevron} colorHex={colorHexChevron} />}
        </>
      )}
    </button>
  );
};

interface LoadingIconProps {
  colorHex?: string;
}

const LoadingIcon: React.FC<LoadingIconProps> = (props) => {
  const { colorHex = '#fff' } = props;

  return (
    <svg
      className={classNames('loadingIcon', styles.loadingIcon)}
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 12C19 13.4555 18.5463 14.8748 17.702 16.0604C16.8577 17.2461 15.6648 18.139 14.2894 18.615C12.9139
        19.0911 11.4242 19.1265 10.0277 18.7164C8.63115 18.3063 7.39717 17.4711 6.49747 16.3269C5.59777 15.1828
        5.0771 13.7866 5.00791 12.3328C4.93872 10.8789 5.32445 9.43961 6.11143 8.2152C6.89841 6.99078 8.04749 6.04214
        9.39877 5.50126C10.7501 4.96039 12.2363 4.85419 13.6508 5.19744"
        stroke={colorHex}
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
  );
};

export default memo(Button);
