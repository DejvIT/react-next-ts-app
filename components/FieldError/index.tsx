import classNames from 'classnames';
import React from 'react';

import styles from './index.module.scss';

interface Props {
  className?: string;
  style?: React.CSSProperties;
  text: string;
}

export const FieldError: React.FC<Props> = (props) => {
  const { text, className, style } = props;

  return (
    <span className={classNames(styles.error, className)} style={style}>
      {text}
    </span>
  );
};
