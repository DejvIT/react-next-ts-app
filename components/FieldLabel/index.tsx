import classNames from 'classnames';
import React from 'react';

import styles from './index.module.scss';

interface Props {
  className?: string;
  style?: React.CSSProperties;
  isRequired?: boolean;
  htmlFor?: string;
  isHorizontal?: boolean;
  onClick?: () => void;
}

export const FieldLabel: React.FC<Props> = (props) => {
  const { htmlFor, className, style, children, isRequired, isHorizontal, onClick } = props;

  return (
    <label
      className={classNames('form-label', styles.label, isHorizontal ? styles.horizontal : null, className)}
      htmlFor={htmlFor}
      onClick={onClick}
      style={style}
    >
      {children}
      {isRequired && <span className="clr-red">*</span>}
    </label>
  );
};
