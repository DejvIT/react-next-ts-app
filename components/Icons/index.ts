export * from './chevronRightIcon';
export * from './fatlessIcon';
export * from './closeIcon';
export * from './ctIcon';
export * from './synapsaIcon';
export * from './mesaIcon';
export * from './facebookIcon';
export * from './junoIcon';
export * from './instagramIcon';
export * from './linkIcon';
export * from './linkedinIcon';
export * from './starIcon';
export * from './twitterIcon';
export * from './denevyIcon';
export * from './youtubeIcon';
