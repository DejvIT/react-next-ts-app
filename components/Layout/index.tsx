import classNames from 'classnames';
import Head from 'next/head';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';
import React, { memo, useMemo } from 'react';
import { Col, Container, Nav, Navbar, NavDropdown, Row } from 'react-bootstrap';

import { CtIcon, DenevyIcon, FatlessIcon, JunoIcon, MesaIcon, StarIcon, SynapsaIcon } from '@app/components';
import { ActionType, useAppState } from '@app/contexts';
import { arrayFromLength, Theme } from '@app/helpers';
import { RouteType, ThemeType } from '@app/types';

import styles from './index.module.scss';

interface Props {
  title: string;
  children: React.ReactNode;
}

export const Layout: React.FC<Props> = (props) => {
  const { title, children } = props;
  const router = useRouter();
  const { appState, dispatch } = useAppState();
  const { t } = useTranslation('common');

  const theme = appState.formData?.theme;

  const rootStyleMemo = useMemo(() => Theme.getRootStyle(theme), [theme]);
  const rootLogoMemo = useMemo(() => Theme.getWebsiteLogo(theme), [theme]);
  const rootFooterMemo = useMemo(() => Theme.getFooterData(theme), [theme]);

  const routes = {
    home: {
      title: t('home.title'),
      url: RouteType.Home(),
    },
    about: {
      title: t('about.title'),
      url: RouteType.About(),
    },
    weather: {
      title: t('weather.title'),
      url: RouteType.Weather(),
    },
    gallery: {
      title: t('gallery.title'),
      url: RouteType.Portfolio(),
    },
    calendar: {
      title: t('calendar.title'),
      url: RouteType.Calendar(),
    },
    contact: {
      title: t('contact.title'),
      url: RouteType.Contact(),
    },
  };

  const partners = [
    {
      title: 'Fatless',
      src: 'https://www.fatless.sk',
      svg: <FatlessIcon colorHex="#ffffff80" />,
    },
    {
      title: 'Denevy',
      src: 'https://www.denevy.eu',
      svg: <DenevyIcon colorHex="#ffffff80" />,
    },
    {
      title: 'Juno One',
      src: 'https://www.juno.one',
      svg: <JunoIcon colorHex="#ffffff80" />,
    },
    {
      title: 'Česká Televize',
      src: 'https://www.ivysilani.cz/',
      svg: <CtIcon colorHex="#ffffff80" />,
    },
    {
      title: 'Synapsa',
      src: 'https://www.synapsa.tech',
      svg: <SynapsaIcon colorHex="#ffffff80" />,
    },
    {
      title: 'Mesa parts',
      src: 'https://www.mesa-parts.cz/',
      svg: <MesaIcon colorHex="#ffffff80" />,
    },
  ];

  const switchTheme = async () => {
    let newTheme: ThemeType | undefined = ThemeType.Primary;
    switch (theme) {
      case ThemeType.Primary:
        newTheme = ThemeType.Secondary;
        break;
      case ThemeType.Secondary:
        newTheme = undefined;
        break;
    }

    dispatch({
      type: ActionType.UpdateTheme,
      payload: {
        formData: {
          theme: newTheme,
        },
      },
    });
  };

  return (
    <>
      <Head>
        <title>{title ? `${title} |` : ''} React App</title>
        <link href="/favicon.ico" rel="icon" />
        <meta content="noindex,nofollow" key="robots" name="robots" />
        <meta content="noindex,nofollow" key="googlebot" name="googlebot" />
      </Head>

      <div>
        <header className={styles.header}>
          <Navbar bg="light" className={styles.navbar} collapseOnSelect expand="lg" variant="light">
            <Container>
              <Navbar.Brand>
                <img
                  alt={rootLogoMemo.alt}
                  className={styles.brandLogo}
                  height={rootLogoMemo.height}
                  src={rootLogoMemo.src}
                  width={rootLogoMemo.width}
                />
              </Navbar.Brand>
              <Navbar.Toggle aria-controls="responsive-navbar-nav" />
              <Navbar.Collapse id="responsive-navbar-nav">
                <Nav className="mr-auto">
                  <Nav.Link onClick={() => router.replace(routes.home.url)}>{t('menu.home')}</Nav.Link>
                  <Nav.Link onClick={() => router.replace(routes.about.url)}>{t('menu.about')}</Nav.Link>
                  <NavDropdown id="collasible-nav-dropdown" title={t('menu.dropdown')}>
                    <NavDropdown.Item onClick={() => router.replace(routes.weather.url)}>
                      {t('menu.weather')}
                    </NavDropdown.Item>
                    <NavDropdown.Item onClick={() => router.replace(routes.gallery.url)}>
                      {t('menu.portfolio')}
                    </NavDropdown.Item>
                    <NavDropdown.Item onClick={() => router.replace(routes.calendar.url)}>
                      {t('menu.calendar')}
                    </NavDropdown.Item>
                    <NavDropdown.Divider />
                    <NavDropdown.Item onClick={() => router.replace(routes.contact.url)}>
                      {t('menu.contact')}
                    </NavDropdown.Item>
                  </NavDropdown>
                </Nav>
              </Navbar.Collapse>
            </Container>
          </Navbar>
          <img alt="car" className={styles.absoluteImage} onClick={switchTheme} src="/img/navbar/car.png" width={250} />
        </header>

        <main className={styles.main}>{children}</main>

        {rootFooterMemo && (
          <>
            <footer className={styles.footer}>
              <Container>
                <div className={styles.footerInsuranceCompanyLogos}>
                  {partners.map((partner, index) => {
                    return (
                      <a href={partner.src} key={index} rel="noreferrer" target="_blank">
                        {partner.svg}
                      </a>
                    );
                  })}
                </div>
                <hr />
                <Row>
                  <Col lg={5}>
                    <img
                      alt="Logo"
                      className={styles.footerLogo}
                      draggable={false}
                      src={rootLogoMemo.src}
                      width={rootLogoMemo.width}
                    />
                    {rootFooterMemo.logoSubtitle && <p>{t(rootFooterMemo.logoSubtitle)}</p>}
                    <ul className={styles.footerSocials}>
                      {rootFooterMemo.socials
                        ?.filter((social) => !!social.url && !!social.type)
                        ?.map((socials) => (
                          <li key={socials.type}>
                            <a
                              href={socials.url}
                              rel="noopener noreferrer"
                              target="_blank"
                              title={socials.type.charAt(0).toUpperCase() + socials.type.slice(1)}
                            >
                              {Theme.getSocialIconForType(socials.type)}
                              <span className="visually-hidden">{socials.type}</span>
                            </a>
                          </li>
                        ))}
                    </ul>
                  </Col>

                  <Col lg={7}>
                    <Row>
                      {rootFooterMemo.sections?.map((section, xIndex) => {
                        return (
                          <Col className={styles.footerSection} key={xIndex} lg={4} sm={6} xs={6}>
                            <strong>{t(section.title)}</strong>
                            <ul>
                              {section?.items
                                ?.filter((item) => !!item.url)
                                ?.map((item, tIndex) => (
                                  <li key={tIndex}>
                                    <a href={item.url as string} target={item.target ? item.target : undefined}>
                                      {t(item.title)} {item.categoryCount ? <span>{item.categoryCount}</span> : null}
                                    </a>
                                  </li>
                                ))}
                            </ul>
                          </Col>
                        );
                      })}
                      {rootFooterMemo.contact && (
                        <Col className={styles.footerSection} lg={4} sm={6} xs={6}>
                          <strong>{t(rootFooterMemo.contact.title)}</strong>
                          <ul className={styles.textStrong}>
                            <li>
                              <span style={{ lineHeight: '1.7rem' }}>{t(rootFooterMemo.contact.streetAddress)},</span>
                              <br />
                              <span style={{ lineHeight: '1.7rem' }}>{t(rootFooterMemo.contact.cityAddress)}</span>
                            </li>
                          </ul>
                          <ul>
                            <li>
                              <span>
                                {t('footer.phone')} {t(rootFooterMemo.contact.phone)}
                              </span>
                            </li>
                            <li>
                              <span>
                                {t('footer.email')}{' '}
                                <a href={`mailto:${t(rootFooterMemo.contact.email)}`}>
                                  {t(rootFooterMemo.contact.email)}
                                </a>
                              </span>
                            </li>
                          </ul>
                          {rootFooterMemo.contact.openingHours && (
                            <ul className={styles.textStrong}>
                              {rootFooterMemo.contact.openingHours?.map((hours, hourIndex) => {
                                return (
                                  <li key={hourIndex}>
                                    <span style={{ width: '80px', display: 'inline-block' }}>{t(hours.interval)}</span>
                                    <span>{hours.time}</span>
                                  </li>
                                );
                              })}
                            </ul>
                          )}
                        </Col>
                      )}
                    </Row>
                  </Col>
                </Row>
                <Row>
                  {rootFooterMemo.rating && (
                    <Col lg={12}>
                      <div className={styles.footerRating}>
                        <div className={styles.stars}>
                          {arrayFromLength(rootFooterMemo.rating?.maxValue)?.map((x) => (
                            <StarIcon isFilled={x + 1 <= Math.round(rootFooterMemo.rating?.value || 0)} key={x} />
                          ))}
                        </div>
                        <div>
                          {rootFooterMemo.rating?.value}
                          <span>
                            {' '}
                            / {rootFooterMemo.rating?.maxValue}
                            {rootFooterMemo.rating?.ratingsCount ? <>・{t('footer.rating.amount')}</> : null}
                          </span>
                        </div>
                      </div>
                      <small className={styles.text}>{t(rootFooterMemo.rating.text as string)}</small>
                    </Col>
                  )}

                  <Col lg={12}>
                    <hr />
                  </Col>

                  <Col className="text-right" lg={12}>
                    <small className={classNames(styles.text, styles.disclaimer)}>
                      2020 - {new Date()?.getFullYear()} {t('footer.disclaimer')}
                    </small>
                  </Col>
                </Row>
              </Container>
            </footer>
          </>
        )}
      </div>
      {rootStyleMemo && (
        <style global jsx key="websiteTheme">{`
          :root {
            ${rootStyleMemo}
          }
        `}</style>
      )}
    </>
  );
};

export default memo(Layout);
