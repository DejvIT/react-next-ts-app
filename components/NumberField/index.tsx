import classNames from 'classnames';
import InputNumber from 'rc-input-number';
import React, { useRef, memo } from 'react';
import { Col, FormGroup, Row } from 'react-bootstrap';

import { FieldError, FieldLabel } from '@app/components';

import styles from './index.module.scss';

interface Props {
  value?: number | null;
  onChange?: (value: number) => void;
  onBlur?: () => void;
  className?: string;
  style?: React.CSSProperties;
  name: string;
  label?: string | JSX.Element;
  placeholder?: string;
  suffix?: string | JSX.Element;

  isRequired?: boolean;
  touched?: boolean;
  error?: string;
  showError?: boolean;

  autocomplete?: string;
  isHorizontal?: boolean;

  disabled?: boolean;

  min?: number;
  max?: number;
  precision?: number;
  decimalSeparator?: string;
}

const FORM_CONTROL_COL = 4;
const FORM_LABEL_COL = 8;

export const NumberField: React.FC<Props> = (props) => {
  const {
    value,
    onChange,
    onBlur,
    suffix,
    label,
    name,
    className,
    style,
    error,
    showError,
    placeholder,
    autocomplete,
    isRequired,
    isHorizontal,
    disabled,
    min,
    max,
    precision,
    decimalSeparator = ',',
  } = props;
  const addonRef = useRef<HTMLSpanElement>(null);
  const addonName = `${name}Addon`;

  return (
    <FormGroup as={Row} className={classNames(styles.formGroup, showError && error ? styles.withError : null)}>
      {label && (
        <Col md={isHorizontal ? FORM_LABEL_COL : 12}>
          <FieldLabel htmlFor={name} isHorizontal={isHorizontal} isRequired={isRequired}>
            {label}
          </FieldLabel>
        </Col>
      )}
      <Col lg={isHorizontal ? FORM_CONTROL_COL : 12}>
        <div className={classNames(styles.inputGroup, suffix ? styles.withAddon : undefined, className)} style={style}>
          <InputNumber
            aria-describedby={suffix ? addonName : undefined}
            aria-label={placeholder}
            autoComplete={autocomplete}
            className={styles.inputNumber}
            decimalSeparator={decimalSeparator}
            disabled={disabled}
            id={name}
            inputMode="tel"
            max={max}
            min={min}
            name={name}
            onBlur={onBlur}
            onChange={onChange}
            placeholder={placeholder}
            precision={precision}
            value={value || undefined}
          />

          {suffix && (
            <span className={styles.addon} id={addonName} ref={addonRef}>
              {suffix}
            </span>
          )}
        </div>

        {showError && error ? <FieldError text={error} /> : null}
      </Col>
    </FormGroup>
  );
};

export default memo(NumberField);
