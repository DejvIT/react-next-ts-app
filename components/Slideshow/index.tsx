import { useRouter } from 'next/router';
import React, { useState, useEffect, memo } from 'react';
import { Nav } from 'react-bootstrap';
import { Carousel, CarouselItem, CarouselControl, CarouselIndicators, CarouselCaption } from 'reactstrap';

import styles from './index.module.scss';

interface Item {
  src: string;
  altText: string;
  caption: string;
  title: string;
  url: string;
}

interface Props {
  rotationLength: number;
  items: Item[];
  showMiniatures: boolean;
  showArrows: boolean;
  height?: number;
  width?: number;
}

export const Slideshow: React.FC<Props> = (props) => {
  const { rotationLength, items, showMiniatures, showArrows, height = 400, width = 800 } = props;
  const router = useRouter();
  const [activeIndex, setActiveIndex] = useState(0);
  const [animating, setAnimating] = useState(false);

  useEffect(() => {
    const interval = setInterval(() => {
      next();
    }, rotationLength * 1000);
    return () => clearInterval(interval);
  });

  const next = () => {
    if (animating) return;
    const nextIndex = activeIndex === items.length - 1 ? 0 : activeIndex + 1;
    setActiveIndex(nextIndex);
  };

  const previous = () => {
    if (animating) return;
    const nextIndex = activeIndex === 0 ? items.length - 1 : activeIndex - 1;
    setActiveIndex(nextIndex);
  };

  const goToIndex = (newIndex: number) => {
    if (animating) return;
    setActiveIndex(newIndex);
  };

  const slides = items.map((item) => {
    return (
      <CarouselItem
        className={styles.customCarouselItem}
        key={item.src}
        onExited={() => setAnimating(false)}
        onExiting={() => setAnimating(true)}
      >
        <img alt={item.altText} className={styles.carouselImage} height={height} src={item.src} width={width} />
        <CarouselCaption captionText={item.caption} className="text-center">
          <h3>{item.caption}</h3>
          <Nav.Link onClick={() => router.replace(item.url)}>{item.title}</Nav.Link>
        </CarouselCaption>
      </CarouselItem>
    );
  });

  return (
    <Carousel activeIndex={activeIndex} next={next} previous={previous}>
      {showMiniatures && <CarouselIndicators activeIndex={activeIndex} items={items} onClickHandler={goToIndex} />}
      {slides}
      {showArrows && (
        <>
          <CarouselControl direction="prev" directionText="Previous" onClickHandler={previous} />
          <CarouselControl direction="next" directionText="Next" onClickHandler={next} />
        </>
      )}
    </Carousel>
  );
};

export default memo(Slideshow);
