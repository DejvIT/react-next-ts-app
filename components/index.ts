export * from './Button';
export * from './Icons';
export * from './Layout';
export * from './Slideshow';
export * from './WeatherCard';
export * from './WeatherDetailModal';
export * from './WeatherTable';
