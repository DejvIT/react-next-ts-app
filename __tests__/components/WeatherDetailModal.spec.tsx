import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { WeatherDetailModal } from "@app/components";
import { unitsMockData, weatherCardMockData } from "@app/__tests__/mock-data/weather-mock";

describe('WeatherDetailModal component', () => {
  it('Should render component when visible', () => {
    const component = shallow(
        <WeatherDetailModal
            isVisible
            onClose={jest.fn()}
            units={unitsMockData}
            weatherData={weatherCardMockData}
        />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Should render component when not visible', () => {
    const component = shallow(
        <WeatherDetailModal
            isVisible={false}
            onClose={jest.fn()}
            units={unitsMockData}
            weatherData={weatherCardMockData}
        />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
