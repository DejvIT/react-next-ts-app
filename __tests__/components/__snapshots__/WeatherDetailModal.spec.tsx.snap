// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetailModal component Should render component when not visible 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  centered={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[MockFunction]}
  restoreFocus={true}
  show={false}
  size="sm"
>
  <ModalHeader
    closeButton={false}
    closeLabel="Close"
  >
    <CloseButton
      aria-label="Close"
      onClick={[MockFunction]}
    />
  </ModalHeader>
  <ModalBody
    className="modalBody"
  >
    <h3
      className="m-mt-24"
    >
      weather.detail.title
    </h3>
    <div>
      <p>
        weather.detail.next1summary
        :
         
        weather.detail.summary.partlycloudy_night
      </p>
      <p>
        weather.detail.next1precipitation
        : 
        0
      </p>
      <p>
        weather.detail.next6summary
        :
         
        weather.detail.summary.cloudy
      </p>
      <p>
        weather.detail.next6precipitation
        : 
        0
      </p>
      <p>
        weather.detail.next12summary
        :
         
        weather.detail.summary.cloudy
      </p>
    </div>
  </ModalBody>
</Modal>
`;

exports[`WeatherDetailModal component Should render component when visible 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  centered={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[MockFunction]}
  restoreFocus={true}
  show={true}
  size="sm"
>
  <ModalHeader
    closeButton={false}
    closeLabel="Close"
  >
    <CloseButton
      aria-label="Close"
      onClick={[MockFunction]}
    />
  </ModalHeader>
  <ModalBody
    className="modalBody"
  >
    <h3
      className="m-mt-24"
    >
      weather.detail.title
    </h3>
    <div>
      <p>
        weather.detail.next1summary
        :
         
        weather.detail.summary.partlycloudy_night
      </p>
      <p>
        weather.detail.next1precipitation
        : 
        0
      </p>
      <p>
        weather.detail.next6summary
        :
         
        weather.detail.summary.cloudy
      </p>
      <p>
        weather.detail.next6precipitation
        : 
        0
      </p>
      <p>
        weather.detail.next12summary
        :
         
        weather.detail.summary.cloudy
      </p>
    </div>
  </ModalBody>
</Modal>
`;
